### DISCUSSION TITLE
How to create a Job using the API that will run only one specific date?

### DISCUSSION THREAD

**Discussion by rafaelsanchezrd** [pschlan/cron-job.org#188] [Q&A]
I'm really impressed by this amazing tool and feel grateful that I discovered it!

I'm currently attempting to use the API to create a Job that will only run once, on a specific date. So far, I've managed to create a Job using the API and here's the payload that I'm sending to the endpoint:

`{
  "job": {
    "url": "https://example.com",
    "enabled": "true",
    "saveResponses": true,
    "title": "Job Feb",
    "schedule": {
      "timezone": "Europe/Berlin",
      "hours": [0],
      "mdays": [24],
      "minutes": [0],
      "months": [2],
      "wdays": [-1]
    }
  }
}`

I'm wondering what steps I should take to ensure that the job runs only once, on a specific date - say, May 31st, 2023. I would greatly appreciate any help or advice on this matter.

**Comment by sokoya** [comment]
I did something like .

while creating the cron Job; you can set a start date and end date in your database 

Check the start date and end date anytime the cron job runs

If the start date matches the date you want then proceed with your code , after then delete the cron job.

I hope this illustration is clear

**Reply by rafaelsanchezrd** [reply]
Thanks for your quick replay:

do you have an example using the API

I sent this to the API and it set a day but also set future dates:

`{"job":{"url":"https://example.com","enabled":"true","saveResponses":true, "body":"check", "title": "03 Infinity list views - Job Feb","schedule": { "timezone": "Europe/Berlin", "hours": [0], "mdays": [25], "minutes": [0], "months": [2], "wdays": [-1] } } }`


how can you set more options like the advanced options using the API?

**Reply by rafaelsanchezrd** [reply]
I would like to set many options like these:

<img width="565" alt="image" src="https://user-images.githubusercontent.com/5248544/221222175-573e9a8f-c54b-4c32-bbca-934f7094d8f2.png">

**Reply by pschlan** [reply]
I'd recommend to check out the data types documented here: https://docs.cron-job.org/rest-api.html#data-types

**Comment by pschlan** [comment]
You can now set an expiry date for your schedule. This should enable you to ensure that the job runs only once (i.e. let it expire immediately after its scheduled first run date).

**Reply by rafaelsanchezrd** [reply]
how to do it via API?

would you please post an example?

**Reply by pschlan** [reply]
In the JobSchedule type, you can set "expiresAt" to a number in format YYYYMMDDhhmmss, e.g. 20230904103000 - see https://docs.cron-job.org/rest-api.html#jobschedule
